File: backend/src/main/java/com/mysite/knitly/domain/mypage/dto/MyPostListItemResponse.java
package com.mysite.knitly.domain.mypage.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import java.time.LocalDateTime;

public record MyPostListItemResponse(
        Long id,
        String title,
        String excerpt,
        String thumbnailUrl,
        @JsonFormat(
                shape = JsonFormat.Shape.STRING,
                pattern = "yyyy-MM-dd'T'HH:mm:ss",
                timezone = "Asia/Seoul"
        )
        LocalDateTime createdAt
) {}

File: backend/src/main/kotlin/com/mysite/knitly/domain/mypage/dto/MyPostListItemResponse.kt
package com.mysite.knitly.domain.mypage.dto

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDateTime

data class MyPostListItemResponse(
    val id: Long,
    val title: String,
    val excerpt: String,
    val thumbnailUrl: String?,
    @JsonFormat(
        shape = JsonFormat.Shape.STRING,
        pattern = "yyyy-MM-dd'T'HH:mm:ss",
        timezone = "Asia/Seoul"
    )
    val createdAt: LocalDateTime
)
